src/mango/src/mango_selector.erl

```diff
@@ -680,7 +697,8 @@ has_required_fields_nested_and_true_test() ->
           ]
     }]},
 
-    ?assertEqual(true, has_required_fields(Selector, RequiredFields)).
+    Normalized = normalize(Selector),
```

Comment:
wups, did I miss this in the last review?

Reply:
Normalization wasn't necessary for the test to pass but I think this makes it clearer that we expect the input to be Normalized. This is because normalized selectors always have an outer `$and` operator so we assume that only `$and` needs to handle the case where it's children are a list and it has no siblings i.e. [here](https://github.com/willholley/couchdb/blob/66510494c8fbadfa86021f82d6f0804d2a92db3a/src/mango/src/mango_selector.erl#L602).

I'm tempted to go further and have a specific record type for normalized selectors (which would also prevent them being normalized multiple times) but that's one for another PR, I think.